### ISSUE TITLE
shim-15.8 for Viasat

### ISSUE THREAD

**Issue by pejdavies1606** [rhboot/shim-review#452]
Confirm the following are included in your repo, checking each box:

 - [x] completed README.md file with the necessary information
 - [x] shim.efi to be signed
 - [x] public portion of your certificate(s) embedded in shim (the file passed to VENDOR_CERT_FILE)
 - [x] binaries, for which hashes are added to vendor_db ( if you use vendor_db and have hashes allow-listed )
 - [x] any extra patches to shim via your own git tree or as files
 - [x] any extra patches to grub via your own git tree or as files
 - [x] build logs
 - [x] a Dockerfile to reproduce the build of the provided shim EFI binaries

*******************************************************************************
### What is the link to your tag in a repo cloned from rhboot/shim-review?
*******************************************************************************
https://github.com/Viasat/shim-review/releases/tag/viasat-shim-x64-20241118

*******************************************************************************
### What is the SHA256 hash of your final SHIM binary?
*******************************************************************************
ad18f6eb26fff69de7d162967512ac3af5accdb9dbbae52152fc76b4a8f8a255

*******************************************************************************
### What is the link to your previous shim review request (if any, otherwise N/A)?
*******************************************************************************
https://github.com/rhboot/shim-review/issues/252

*******************************************************************************
### If no security contacts have changed since verification, what is the link to your request, where they've been verified (if any, otherwise N/A)?
*******************************************************************************
https://github.com/rhboot/shim-review/issues/252

**Comment by steve-mcintyre** [comment]
Contacts have previously been verified in #252

**Comment by dennis-tseng99** [comment]
I tried to build shim binary, but I'm sorry I cannot find the Dockerfile. Although in your README.md, you have already showed the build procedure, is it possible to build by ```docker build .``` ? Why do you use that case ? Thanks.

**Comment by pejdavies1606** [comment]
> I tried to build shim binary, but I'm sorry I cannot find the Dockerfile. Although in your README.md, you have already showed the build procedure, is it possible to build by `docker build .` ? Why do you use that case ? Thanks.

Docker is no longer available due to licensing restrictions.

**Comment by pejdavies1606** [comment]
New tag
- https://github.com/Viasat/shim-review/releases/tag/viasat-shim-x64-20250218

New SHA256 hash
- `ac0c115c3ab7f764e05db7213f772ece8b58a06566a6e5f70d8eddbc756f342a`

Summary of changes
- Use Ubuntu release instead of daily
  - Daily is no longer available.
  - Update pinned git version.
  - Remove text encoding assumption from subprocess output.
- Answer question about kernel lockdown
  - No Linux kernel.

**Comment by pejdavies1606** [comment]
Yes, we use an EV certificate. This review submission is to update the secure boot shim for a legacy product that is still signed with the current certificate to address customers affected by https://github.com/rhboot/shim/issues/682 since the infamous [Windows Update on August 13 2024](https://support.microsoft.com/en-us/topic/august-13-2024-kb5041571-os-build-26100-1457-d218c08d-8de2-4f9a-8fe1-a2c2fd83ca9a).

**Comment by ntgrai** [comment]
Considering all people reviews are volunteers and are likely 100% linux users requiring us to have a WSL-capable machine is unrealistic or have your build process structured around modifying the host system (you currently install some packages). If you want your review to be processed in a sensible time frame you should make all possible efforts to simplify the work of reviewers.

**Comment by pejdavies1606** [comment]
New tag
- https://github.com/Viasat/shim-review/releases/tag/viasat-shim-x64-20250416

New SHA256 hash
- `5868bc3c0afab377b1d622e571a82a780277490449d02bd49bcf77e9bbfad207`

Summary of changes
- Answer new question in README about CA certificate.
- Remain at shim-15.8.
- Restore Dockerfile.

**Comment by pejdavies1606** [comment]
Any news on this please?

**Comment by steve-mcintyre** [comment]
It appears that your embedded certificate has expired. Do you have a plan to update that?

**Comment by pejdavies1606** [comment]
New tag

- https://github.com/Viasat/shim-review/releases/tag/viasat-shim-x64-20250701

New hash

- `378743b24f76d5ae8c0ba83af03ae26b9f27c75629171bbef3fec2a4c46168b4`

Summary of changes

- Update to shim 16.0
- Update certificate
- Update Dockerfile based on example with pinned packages
- Address executable flag of `check_nx_bit.sh`

**Comment by steve-mcintyre** [comment]
Ping? We're waiting on a response here.

**Comment by pejdavies1606** [comment]
It seems `software-properties-common` v0.99.49.2 has been [superceded](https://launchpad.net/ubuntu/noble/amd64/software-properties-common/0.99.49.2) and this has broken the reproducible build, resulting in a different checksum.

We are not loading a Linux kernel, so it follows that we will not be loading UKIs, but we might as well update to shim 16.1 and try again anyway.

**Comment by marta-lewandowska** [comment]
- building shim 16.1 with no additional patches
- not loading GRUB, sd-boot, kernel, UKIs
- shim only loads PAE (Pre-Authentication Environment) application
- 4096 bit cert was updated, as noted before, and it is EV not CA
- I ran build.sh and only changed the `docker` commands to `podman` and I'm still getting different sha sums
```
STEP 8/10: RUN sha256sum _viasat/INSTALL/shimx64.efi shimx64.efi
fabae159adf03a93e028f3cee187b59c094c8036f32db752b2ec25a2d224976a  _viasat/INSTALL/shimx64.efi
b8da347cd60df24f778b71557fe0846c1b095d7d08f71ac6d19682a571d7ef9d  shimx64.efi
```

Everything else looks ok to me.

**Comment by marta-lewandowska** [comment]
Hi, I just ran it again, to be sure. Here's a link to the logs: https://github.com/marta-lewandowska/viasat_shim-review

**Comment by evilteq** [comment]
I also tried to replicate, from viasat-shim-x64-20251027, I get the same sha256 as Marta.

**Comment by marta-lewandowska** [comment]
```
STEP 8/10: RUN sha256sum _viasat/INSTALL/shimx64.efi shimx64.efi
fabae159adf03a93e028f3cee187b59c094c8036f32db752b2ec25a2d224976a  _viasat/INSTALL/shimx64.efi
fabae159adf03a93e028f3cee187b59c094c8036f32db752b2ec25a2d224976a  shimx64.efi
```
Ok, looks like we're done. That was the last outstanding issue. Thanks for working through this. :)
